In [1]:
%run ../aux/postgres.py

postgres functions loaded...


In [10]:
query("select * from amostra where lower(descricao) like '%covid%'")

[(222, 'Covid-19 PT', None, None, None, None),
 (223, 'Covid-19 FR', None, None, None, None)]

~~~~sql
select count(codtweet) from amostratweet where codamostra = 222
~~~~
12.865.843

~~~~sql
select count(codtweet) from amostratweet where codamostra = 223
~~~~
11.861.867

In [9]:
execSQL("insert into amostra (descricao) values ('Covid-19 FR')")

In [ ]:
print('a')

In [1]:
!ls data

corona1k.rds			    tweetsfr.txt
corona.rds			    tweets_per_user.csv
count_max_tweets_user_day_lang.csv  tweets_per_user_day.csv
max_tweets_user_day_lang.csv	    tweetspt.txt
tweets1k.csv			    tweets_user_day_lang.csv


## Exportando tweets:

~~~~
psql -d twitter -t -c "select json from tweet where codprojeto = 37 and datatweet between '2020-03-20' and '2020-04-09' and json->>'lang' = 'pt'"  > tweetspt.txt

psql -d twitter -t -c "select json from tweet where codprojeto = 37 and datatweet between '2020-03-20' and '2020-04-09' and json->>'lang' = 'fr'"  > tweetsfr.txt

mateus@kode:~/projetos/absa/20_corona/data$ wc -l tweetspt.txt
12.865.844 tweetspt.txt

mateus@kode:~/projetos/absa/20_corona/data$ wc -l tweetsfr.txt
11.861.868 tweetsfr.txt
~~~~

## SQL conjuntos com tweets em português e francês:

~~~~sql
INSERT INTO amostratweet (codamostra, codtweet) select 222, codtweet from tweet where codprojeto = 37 and datatweet between '2020-03-20' and '2020-04-09' and json->>'lang' = 'pt';

insert into amostratweet (codamostra, codtweet) select 223, codtweet from tweet where codprojeto = 37 and datatweet between '2020-03-20' and '2020-04-09' and json->>'lang' = 'fr';
~~~~

## Tweets por usuário por dia:

~~~~
psql -d twitter -t -c "SELECT userid, datatweet, count(t.codtweet) as tweets FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE at.codamostra = 222 GROUP BY userid, datatweet" > tweets_user_day_pt.csv

psql -d twitter -t -c "SELECT userid, datatweet, count(t.codtweet) as tweets FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE at.codamostra = 223 GROUP BY userid, datatweet" > tweets_user_day_fr.csv

~~~~


In [13]:
query("select * from deteccaobot where descricao like '%pt'")

[(1, 'Tweets por dia pt'),
 (4, 'Default Profile Image pt'),
 (6, 'Followers count pt'),
 (8, 'Friends count pt'),
 (10, 'Created at pt'),
 (12, 'Group by 3 pt'),
 (13, 'Group by 4 pt')]

In [14]:
query("select * from deteccaobot where descricao like '%fr'")

[(2, 'Tweets por dia fr'),
 (5, 'Default Profile Image fr'),
 (7, 'Followers count fr'),
 (9, 'Friends count fr'),
 (11, 'Created at fr'),
 (14, 'Group by 3 fr'),
 (15, 'Group by 4 fr')]

In [4]:
select userid, count(userid) from deteccaobotuserid where coddeteccaobot in (1, 4, 6, 8, 10) order by 1 group by 1

[]

In [ ]:
select userid, count(userid) from deteccaobotuserid where coddeteccaobot in (2, 5, 7, 9, 11) order by 1 group by 1

~~~~sql
insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 1, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE at.codamostra = 222 GROUP BY userid, datatweet having count(t.codtweet) >= 100;
insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 2, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE at.codamostra = 223 GROUP BY userid, datatweet having count(t.codtweet) >= 100;


~~~~

## Imagem

In [ ]:
tweet['user']['default_profile_image']

In [8]:
query("select * from deteccaobot where descricao like 'Default Profile%'")

[(4, 'Default Profile Image pt'), (5, 'Default Profile Image fr')]

222 - português
223 - francês
~~~~sql
insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 4, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE at.codamostra = 222 AND json->'user'->>'default_profile_image' = 'false';
insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 5, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE at.codamostra = 223 AND json->'user'->>'default_profile_image' = 'false';

~~~~

In [ ]:
query("select coddeteccaobot, count(*) from deteccaobotuserid group by 1")

In [ ]:
execSQL("insert into deteccaobot (descricao) values ('Followers count pt')")

In [ ]:
execSQL("insert into deteccaobot (descricao) values ('Followers count fr')")

In [ ]:
friends_count

In [12]:
execSQL("insert into deteccaobot (descricao) values ('Group by 3 pt')")
execSQL("insert into deteccaobot (descricao) values ('Group by 4 pt')")

execSQL("insert into deteccaobot (descricao) values ('Group by 3 fr')")
execSQL("insert into deteccaobot (descricao) values ('Group by 4 fr')")

In [ ]:
execSQL("insert into deteccaobot (descricao) values ('Friends count fr')")

In [ ]:
execSQL("insert into deteccaobot (descricao) values ('Created at pt')")

In [ ]:
print('a')

~~~~sql
--- followers count
insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 6, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE t.codprojeto = 37 and at.codamostra = 222 AND json->'user'->>'followers_count' = '0';

insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 7, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE t.codprojeto = 37 and at.codamostra = 223 AND json->'user'->>'followers_count' = '0';  

--- friends count
insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 8, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE t.codprojeto = 37 and at.codamostra = 222 AND json->'user'->>'friends_count' = '0';

insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 9, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE t.codprojeto = 37 and at.codamostra = 223 AND json->'user'->>'friends_count' = '0';  

--- created at
insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 10, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE t.codprojeto = 37 and at.codamostra = 222 AND ((json->>'created_at')::date - (json->'user'->>'created_at')::date) <=7;

insert into deteccaobotuserid (coddeteccaobot, userid) SELECT distinct 11, userid FROM tweet t INNER JOIN amostratweet at ON at.codtweet = t.codtweet WHERE t.codprojeto = 37 and at.codamostra = 223 AND ((json->>'created_at')::date - (json->'user'->>'created_at')::date) <= 7;  



In [17]:
query("select * from deteccaobot where descricao like 'Group%'")

[(12, 'Group by 3 pt'),
 (13, 'Group by 4 pt'),
 (14, 'Group by 3 fr'),
 (15, 'Group by 4 fr')]

~~~~sql
insert into deteccaobotuserid (coddeteccaobot, userid) SELECT 12, userid from deteccaobotuserid where coddeteccaobot in (1, 4, 6, 8, 10) group by userid having count(*) >= 3;

insert into deteccaobotuserid (coddeteccaobot, userid) SELECT 13, userid from deteccaobotuserid where coddeteccaobot in (1, 4, 6, 8, 10) group by userid having count(*) >= 4;

insert into deteccaobotuserid (coddeteccaobot, userid) SELECT 14, userid from deteccaobotuserid where coddeteccaobot in (2, 5, 7, 9, 11) group by userid having count(*) >= 3;

insert into deteccaobotuserid (coddeteccaobot, userid) SELECT 15, userid from deteccaobotuserid where coddeteccaobot in (2, 5, 7, 9, 11) group by userid having count(*) >= 4;
~~~~

In [24]:
query("SELECT dbu.coddeteccaobot, db.descricao, count(*) as users FROM deteccaobotuserid dbu INNER JOIN deteccaobot db ON db.coddeteccaobot = dbu.coddeteccaobot WHERE db.descricao not like 'Group%' AND db.descricao like '%pt' GROUP BY 1, 2")

[(1, 'Tweets por dia pt', 297),
 (4, 'Default Profile Image pt', 2203618),
 (6, 'Followers count pt', 38462),
 (8, 'Friends count pt', 9034),
 (10, 'Created at pt', 57924)]

In [26]:
query("SELECT dbu.coddeteccaobot, db.descricao, count(*) as users FROM deteccaobotuserid dbu INNER JOIN deteccaobot db ON db.coddeteccaobot = dbu.coddeteccaobot WHERE db.descricao like 'Group%' AND db.descricao like '%pt' GROUP BY 1, 2")

[(12, 'Group by 3 pt', 19452), (13, 'Group by 4 pt', 2983)]

In [25]:
query("SELECT dbu.coddeteccaobot, db.descricao, count(*) as users FROM deteccaobotuserid dbu INNER JOIN deteccaobot db ON db.coddeteccaobot = dbu.coddeteccaobot WHERE db.descricao not like 'Group%' AND db.descricao like '%fr' GROUP BY 1, 2")

[(2, 'Tweets por dia fr', 653),
 (5, 'Default Profile Image fr', 1340381),
 (7, 'Followers count fr', 35446),
 (9, 'Friends count fr', 10138),
 (11, 'Created at fr', 34767)]

In [27]:
query("SELECT dbu.coddeteccaobot, db.descricao, count(*) as users FROM deteccaobotuserid dbu INNER JOIN deteccaobot db ON db.coddeteccaobot = dbu.coddeteccaobot WHERE db.descricao like 'Group%' AND db.descricao like '%fr' GROUP BY 1, 2")

[(14, 'Group by 3 fr', 17430), (15, 'Group by 4 fr', 3721)]